src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1666,6 +1672,180 @@ public void visitSwitchExpression(JCTree.JCSwitchExpression tree) {
+                            }
+
+                            append(CoreOp._yield(labelResult));
+                            clBodies.add(stack.body);
+
+                            // Pop label
+                            popBody();
+                        }
+
+                        localResult = append(ExtendedOp.conditionalOr(clBodies));
+                    }
+
+                    append(CoreOp._yield(localResult));
+                    bodies.add(stack.body);
+
+                    // Pop labels
+                    popBody();
+                } else if (headCl instanceof JCTree.JCDefaultCaseLabel) {
+                    // @@@ Do we need to model the default label body?
+                    pushBody(headCl, FunctionType.VOID);
+
+                    append(CoreOp._yield());
+                    bodies.add(stack.body);
+
+                    // Pop label
+                    popBody();
+                } else {
+                    throw unsupported(tree);
+                }
+
+                // Statements body
+                switch (c.caseKind) {
+                    case RULE -> {
+                        pushBody(c.body, actionType);
+                        if (c.body instanceof JCTree.JCBlock b) {
+                            toValue(b);
+                            if (!(b.stats.last() instanceof JCTree.JCBreak)) {
+                                append(CoreOp._yield()); // @@@ _break is also an option
+                            }
+                        }
+                        else if (c.body instanceof JCTree.JCStatement s) {
+                            toValue(s);
+                            if (!(s instanceof JCTree.JCThrow)) {
+                                append(CoreOp._yield());
+                            }
+                        }
+                        bodies.add(stack.body);
+
+                        // Pop block
+                        popBody();
+                    }
+                    case STATEMENT -> {
+                        // @@@ Avoid nesting for a single block? Goes against "say what you see"
+                        // boolean oneBlock = c.stats.size() == 1 && c.stats.head instanceof JCBlock;
+                        pushBody(c, actionType);
+
+                        scan(c.stats);
+
+                        appendTerminating(c.completesNormally ?
+                                headCl instanceof JCTree.JCDefaultCaseLabel ? CoreOp::_yield : ExtendedOp::switchFallthroughOp
+                                : CoreOp::unreachable);
+
+                        bodies.add(stack.body);
+
+                        // Pop block
+                        popBody();
+                    }
+                };
+            }
+
+            // if enhanced switch and no default label
+            // enhanced: target type not byte, char short, int, String, or, has case pattern or case null
+            boolean enhancedSw = !List.of(JavaType.BYTE, JavaType.CHAR, JavaType.SHORT, JavaType.INT,
```

Comment:
In `Lower` the code corresponding to this is:
```
public void visitSwitch(JCSwitch tree) {
        List<JCCase> cases = tree.patternSwitch ? addDefaultIfNeeded(tree.patternSwitch,
                                                                     tree.wasEnumSelector,
                                                                     tree.cases)
                                                : tree.cases;
        handleSwitch(tree, tree.selector, cases);
    }
```

So, I don't think you need all this state tracking and you can probably simplify all this.